src/main/java/ui/UI.java

```diff
@@ -595,4 +598,42 @@ private void showJavaRuntimeVersionNotCompatible(String javaRuntimeVersionString
         String message = String.format(ERROR_MSG_JAVA_RUNTIME_VERSION_PARSING, javaRuntimeVersionString);
         DialogMessage.showInformationDialog("Java version unknown", message);
     }
+
+     /* This function updates the TickerTimer period, which is used to periodically refresh the issues.
+     * @param apiQuota : The remaining allowed api request until the next api request allowance renewal.
+     * @param remainingTime : The remaining time left until the next api request allowance renewal.
+     * @param lastApiCallsUsed : The amount of api used in the last api pull.
+     */
+    public int updateTickerTimer(int apiQuota, long remainingTime, int lastApiCallsUsed){
+
+        long refreshTimeInMins;
+        if (lastApiCallsUsed > 0) {
+            // If less or equal than 200, restricts auto refresh until next apiQuota renewal.
+            // The 200 apiQuota allowance is reserved for Manual refresh and creation of issues, etc.
+            if (apiQuota <= 200){
+                refreshTimeInMins = remainingTime + 1;
+            } else {
+                if (apiQuota - 200 >= lastApiCallsUsed) {
+                    refreshTimeInMins =  (long) Math.ceil(remainingTime /
+                            Math.floor((apiQuota - 200) / (double) lastApiCallsUsed));
+                } else {
+                    refreshTimeInMins = remainingTime + 1; // Restricts auto refresh until next apiQuota renewal.
+                }
+            }
+        } else { // Enters here during application initialisation or when apiQuota = 0 and lastApiCallsUsed = 0
+            if (apiQuota != 0) {
+                refreshTimeInMins = REFRESH_PERIOD / 60; // Set to default during initialisation
+            } else {
+                refreshTimeInMins = remainingTime + 1; // Restricts auto refresh until next apiQuota renewal.
+            }
+        }
+
+        refreshTimer.pause();
```

Comment:
Do look at the TickingTimer header comment for how to synchronise compound operations.

Also, maybe this can be made a synchronised method in TickingTimer instead. Higher level of abstraction than just setting the period.